tools/format_python_tools.py

```diff
@@ -12,19 +12,20 @@
 EXCLUDE_FILES = ['gen_compilation_database.py']
 
 
-def collectFiles():
+def collectFiles(directory):
   """Collect all Python files in the tools directory.
 
+  Args:
+    directory: A string, path to a directory where to look for files that should
+      be checked for correct formatting.
+
   Returns: A collection of python files in the tools directory excluding
     any directories in the EXCLUDE_DIRECTORIES constant.
   """
   # TODO: Add ability to collect a specific file or files.
   matches = []
   path_parts = os.getcwd().split('/')
-  dirname = '.'
-  if path_parts[-1] == 'tools':
-    dirname = '/'.join(path_parts[:-1])
-  for root, dirnames, filenames in os.walk(dirname):
+  for root, dirnames, filenames in os.walk(directory):
```

Comment:
as a general point `pathlib` is a nicer/more explicit interface to work with

i resisted for a long time being old-school, but it really does improve things

Reply:
I agree. I don't think it is important to change it in this PR though. Please let me know if you disagree.

Reply:
no - totally general point